applications/dashboard/controllers/class.activitycontroller.php

```diff
@@ -237,13 +237,13 @@ public function delete($ActivityID = '', $TransientKey = '') {
 
         $this->ActivityModel->delete($ActivityID);
 
-        if ($this->_DeliveryType === DELIVERY_TYPE_ALL) {
-            redirect(GetIncomingValue('Target', $this->SelfUrl));
+        $target = Gdn::request()->get('Target');
+        if ($this->_DeliveryType === DELIVERY_TYPE_ALL && $target) {
+            redirect($target);
         }
 
-        // Still here? Getting a 404.
-        $this->ControllerName = 'Home';
-        $this->View = 'FileNotFound';
+        // Still here? Getting a basic view with redirect.
+        $this->RedirectUrl = url('activity');
```

Comment:
No; the concern was that if, thru a failure of Javascript, you manage to actually land on the delete activity page, you'll then be stuck there. Obviously that doesn't justify a crappy default experience.